tests/poll.rs

```diff
@@ -201,6 +208,94 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+#[test]
+fn registry_operations_are_thread_safe() {
+    init();
+
+    let (mut poll, mut events) = init_with_poll();
+
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+    let registry1 = Arc::clone(&registry);
+    let registry2 = Arc::clone(&registry);
+    let registry3 = Arc::clone(&registry);
+
+    let barrier = Arc::new(Barrier::new(4));
+    let barrier1 = Arc::clone(&barrier);
+    let barrier2 = Arc::clone(&barrier);
+    let barrier3 = Arc::clone(&barrier);
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+    let addr = listener.local_addr().unwrap();
+
+    let mut udp_socket = UdpSocket::bind(any_local_address()).unwrap();
```

Comment:
Small thing; this can be moved into the `thread::spawn`.

Reply:
ok, moved.